plugins/language_go.lua

```diff
@@ -6,20 +6,21 @@ syntax.add {
   files = { "%.go$" },
   comment = "//",
   patterns = {
-    { pattern = "//.-\n",               type = "comment"  },
-    { pattern = { "/%*", "%*/" },       type = "comment"  },
-    { pattern = { '"', '"', '\\' },     type = "string"   },
-    { pattern = { "`", "`", '\\' },     type = "string"   },
-    { pattern = { "'", "'", '\\' },     type = "string"   },
-    { pattern = "0[oO_][0-7]+",         type = "number"   },
-    { pattern = "-?0x[%x_]+",           type = "number"   },
-    { pattern = "-?%d+_%d",             type = "number"   },
-    { pattern = "-?%d+[%d%.eE]*f?",     type = "number"   },
-    { pattern = "-?%.?%d+f?",           type = "number"   },
-    { pattern = "[%+%-=/%*%^%%<>!~|&]", type = "operator" },
-    { pattern = ":=",                   type = "operator" },
-    { pattern = "[%a_][%w_]*%f[(]",     type = "function" },
-    { pattern = "[%a_][%w_]*",          type = "symbol"   },
+    { pattern = "//.-\n",                            type = "comment"  },
+    { pattern = { "/%*", "%*/" },                    type = "comment"  },
+    { pattern = { '"', '"', '\\' },                  type = "string"   },
+    { pattern = { "`", "`", '\\' },                  type = "string"   },
+    { pattern = { "'", "'", '\\' },                  type = "string"   },
+    { pattern = "0[oO_][0-7]+",                      type = "number"   },
+    { pattern = "-?0x[%x_]+",                        type = "number"   },
+    { pattern = "-?%d+_%d",                          type = "number"   },
+    { pattern = "-?%d+[%d%.eE]*f?",                  type = "number"   },
+    { pattern = "-?%.?%d+f?",                        type = "number"   },
+    { pattern = "[%+%-=/%*%^%%<>!~|&]",              type = "operator" },
+    { pattern = ":=",                                type = "operator" },
+    { pattern = "[%a_][%w_]*%f[(]",                  type = "function" }, -- function call
+    { pattern = "func()[%s].*[%a_][%w_]*()%f[%[(]",  type = {"keyword", "function", "normal"} }, -- function statement
```

Comment:
`[%s].*` would match a space followed by anything. To only match spaces use this:
```suggestion
    { pattern = "func()%s*[%a_][%w_]*()%f[%[(]",  type = {"keyword", "function", "normal"} }, -- function statement
```